### PR TITLE
Updated to work with Webpack v2.2.0-rc.2

### PR DESCRIPTION
**Pull request by PizzaPete**
Changed module.entry into module.entryModule.
Changed chunk.initial into chunk.isInitial()

### CODE REVIEW

**Comment by PizzaPete** [comment]
This will probably break on Webpack 1.x.

File: src/webpack-system-register.js [hunk 1]
```diff
@@ -114,7 +114,7 @@ WebpackSystemRegister.prototype.apply = function(compiler) {
 		// http://stackoverflow.com/questions/35092183/webpack-plugin-how-can-i-modify-and-re-parse-a-module-after-compilation
 		compilation.plugin('seal', () => {
 			compilation.modules.forEach(module => {
-				let isEntry = module.entry;
+				let isEntry = module.entryModule;
```
Comment (internettrans) [inline src/webpack-system-register.js:117]: Could we get this to work with both webpack 1 and webpack 2? Maybe `module.entry || module.entryModule`

File: src/webpack-system-register.js [hunk 2]
```diff
@@ -131,7 +131,7 @@ WebpackSystemRegister.prototype.apply = function(compiler) {
 		// Based on https://github.com/webpack/webpack/blob/ded70aef28af38d1deb2ac8ce1d4c7550779963f/lib/WebpackSystemRegister.js
 		compilation.plugin("optimize-chunk-assets", (chunks, callback) => {
 			chunks.forEach(chunk => {
-				if (!chunk.initial) {
+				if (!chunk.isInitial()) {
```
Comment (internettrans) [inline src/webpack-system-register.js]: Same here about trying to keep it backwards compatible. Maybe something like this: `chunk.isInitial ? !chunk.isInitial() : !chunk.initial`?

**Review by internettrans — COMMENTED** [review]
Thanks for your help with this - compatibility with webpack 2 will be awesome. I'm hoping we can get it working in both webpack 1 and webpack 2, though. What are your thoughts?

**Comment by PizzaPete** [comment]
@joeldenning I will check to see if we can keep it backwards compatible.
Just after my pull request I realised that this breaks Webpack 1.

File: src/webpack-system-register.js (hunk 1 above)
Comment (PizzaPete) [reply src/webpack-system-register.js:117]: Tried `module.entryModule` with Webpack 1.13.0, Webpack 1.14.0 (latest) and Webpack 2.2.0-rc.2.
Seems to work for all the versions.

File: src/webpack-system-register.js (hunk 2 above)
Comment (PizzaPete) [reply src/webpack-system-register.js]: Indeed, changes this into the code you suggested and it works for Webpack 1.13.0, Webpack 1.14.0 (latest) and Webpack 2.2.0-rc.2.

**Comment by PizzaPete** [comment]
@joeldenning I've tested the webpack-app example within the example directory.

The systemjs-app example doesn't work by just running `npm install` and `npm run bundle`.

**Review by internettrans — APPROVED** [review]
Looks good 👍 

Just tested this out with the example and also with another Canopy project that depends on this -- both worked great.

Thanks for you help with this!

**Comment by internettrans** [comment]
Released as [1.4.0](https://github.com/CanopyTax/webpack-system-register/releases/tag/v1.4.0)

**Pull request merged**